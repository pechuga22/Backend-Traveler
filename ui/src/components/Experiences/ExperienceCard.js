import React from 'react'
import { Comment, Icon, Rating, Image } from 'semantic-ui-react'
import moment from 'moment'

import './ExperienceCard.less'
import Resources from '../Resources/Resources'

export default function ExperienceCard({ experience }) {
  moment.locale('en', {
    relativeTime: {
      future: 'in %s',
      past: '%s ago',
      s: 's',
      m: '1 min',
      mm: '%d min',
      h: '1 h',
      hh: '%d h',
      d: '1 d',
      dd: '%d d',
      M: '1 m',
      MM: '%d m',
      y: '1 y',
      yy: '%d y',
    },
  })
  const {
    _id,
    avatar,
    name,
    creationDate,
    rating,
    content,
    resources,
  } = experience
  return (
    <Comment.Group className='recomendations'>
      <Comment id={_id}>
        <Image
          src={avatar}
          floated='left'
          circular
          rounded
          size='mini'
          avatar
          bordered
        />
        <Comment.Content>
          <Comment.Author as='a'>{name}</Comment.Author>
          <Comment.Metadata>
            <Icon className='clock outline' />
            <div>{moment(creationDate).fromNow()}</div>
          </Comment.Metadata>
          <div>
            <Rating icon='favorite' defaultRating={rating} maxRating={rating} />
          </div>
        </Comment.Content>
        <Comment.Text>{content}</Comment.Text>
        {resources ? <Resources resources={resources} /> : null}
      </Comment>
    </Comment.Group>
  )
}
