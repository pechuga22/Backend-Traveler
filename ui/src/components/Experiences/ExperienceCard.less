@import '../../variables.less';

.ui.recomendations {
  .comment {
    a.author {
      color: @primaryColor;
      font-size: 16px;
      text-transform: capitalize;
    }

    div.text {
      color: @secondaryColor;
      margin-top: 1.5rem;
    }

    .ui.bordered.circular.left.floated.image.avatar {
      height: 3.5rem;
      width: 3.5rem;
      object-fit: contain;
      border-radius: 50%;
      margin-right: 0.5rem;
    }

    div.metadata {
      float: right;
    }

    div.text {
      margin-top: 0.5rem;
    }
  }
}
