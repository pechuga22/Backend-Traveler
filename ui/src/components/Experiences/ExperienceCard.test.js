import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ExperienceCard from './ExperienceCard'
import { Image } from 'semantic-ui-react'

const experience = {
  avatar:
    'https://lh5.googleusercontent.com/-nfK-6KaZMRg/AAAAAAAAAAI/AAAAAAAAAAA/AMZuucmGlEjmzOXQl1tS5svUBgOqLfcXRw/photo.jpg',
  content:
    'Vocês passam no Cristo de la Concórdia? Tem disponibilidade para o dia 18/08? E vocês pegam passageiros na rodoviária?',
  creationDate: '2020-06-14T11:40:22.227000+00:00',
  rating: 4,
  resources: [
    { type: 'video', source: 'https://www.youtube.com/watch?v=E_WQOBXE7us' },
  ],
  name: 'Albeiro Crespo',
  _id: '5eecd780c9df78c86a6ct541x',
}

describe('./components/Experiences/ExperienceCard.js', () => {
  describe('.render()', () => {
    let component

    beforeEach(() => {
      component = shallow(
        <ExperienceCard experience={experience} key={experience._id} />
      )
    })


    it('should render the ExperienceCard component', () => {
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should find class into ExperienceCard component', () => {
      const wrapper = component.find('.clock')
      expect(wrapper.length).toBe(1)
    })

    it('should find avatar into ExperienceCard component', () => {
      const { avatar } = experience
      const img = <Image src={avatar} />
      expect(component.containsMatchingElement(img)).toBe(true)
    })

    it('should validate resource is not empty', () => {
      const { resources } = experience
      expect(resources.length > 0).toEqual(true)
    })

    it('should validate resource is empty', () => {
      const experienceResource = {
        _id: '5eecd780c9df78c86a6ct541x',
        avatar:
          'https://lh5.googleusercontent.com/-nfK-6KaZMRg/AAAAAAAAAAI/AAAAAAAAAAA/AMZuucmGlEjmzOXQl1tS5svUBgOqLfcXRw/photo.jpg',
        creationDate: '2020-06-14T11:40:22.227000+00:00',
        resources: []
      }
      const {resources} = experienceResource
      expect(resources.length===0).toBe(true)
    })
  })
})
