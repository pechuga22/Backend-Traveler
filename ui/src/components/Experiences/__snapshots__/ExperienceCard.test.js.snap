// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Experiences/ExperienceCard.js .render() should render the ExperienceCard component 1`] = `
<CommentGroup
  className="recomendations"
>
  <Comment
    id="5eecd780c9df78c86a6ct541x"
  >
    <Image
      as="img"
      avatar={true}
      bordered={true}
      circular={true}
      floated="left"
      rounded={true}
      size="mini"
      src="https://lh5.googleusercontent.com/-nfK-6KaZMRg/AAAAAAAAAAI/AAAAAAAAAAA/AMZuucmGlEjmzOXQl1tS5svUBgOqLfcXRw/photo.jpg"
      ui={true}
    />
    <CommentContent>
      <CommentAuthor
        as="a"
      >
        Albeiro Crespo
      </CommentAuthor>
      <CommentMetadata>
        <Icon
          as="i"
          className="clock outline"
        />
        <div>
          25 d ago
        </div>
      </CommentMetadata>
      <div>
        <Rating
          clearable="auto"
          defaultRating={4}
          icon="favorite"
          maxRating={4}
        />
      </div>
    </CommentContent>
    <CommentText>
      Vocês passam no Cristo de la Concórdia? Tem disponibilidade para o dia 18/08? E vocês pegam passageiros na rodoviária?
    </CommentText>
    <Resources
      resources={
        Array [
          Object {
            "source": "https://www.youtube.com/watch?v=E_WQOBXE7us",
            "type": "video",
          },
        ]
      }
    />
  </Comment>
</CommentGroup>
`;
